File: app/javascript/channels/acns1/room_channel.js
// import consumer from '../consumer'
//
//   const chatChannel = consumer.subscriptions.create({ channel: 'Acns1::RoomChannel', room_id: $('#messages').data('room_id') }, {
//     connected() {
//       // Called when the subscription is ready for use on the server
//     },
//
//     disconnected() {
//       // Called when the subscription has been terminated by the server
//     },
//
//     received(data) {
//       return $('#messages').append(data['message'])
//     },
//
//     speak(message) {
//       return this.perform('speak', { message: message })
//     }
//   })
//
//   $(document).on('keypress', '[data-behavior~=room_speaker]', function(event) {
//     if (event.keyCode === 13) {
//       chatChannel.speak(event.target.value)
//       event.target.value = ''
//       return event.preventDefault()
//     }
//   })
// })

File: app/channels/acns1/room_channel.rb
class Acns1::RoomChannel < ApplicationCable::Channel
  def subscribed
    p ["#{__FILE__}:#{__LINE__}", __method__, params]
    stream_from "acns1/room_channel/#{params['room_id']}"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    Acns1::Message.create!(body: data['message'], user: current_user, room_id: params['room_id'])
  end
end
